### ISSUE TITLE
Skeleton Archers Don't Drop Materials When Exploded

### ISSUE THREAD

**Issue by Zeinath** [jpw1991/chebs-necromancy#224]
# Description

With config set to drop everything, tested with the drop to container enabled and disabled, when a skeleton archer (fire) dies or get whacked with the wand, they explode in the death animation, but no resources appear to be dropped in any form.

## Attachments Checklist

[LogOutput.log](https://github.com/jpw1991/chebs-necromancy/files/11876866/LogOutput.log)

[com.chebgonaz.ChebsNecromancy.cfg.txt](https://github.com/jpw1991/chebs-necromancy/files/11876871/com.chebgonaz.ChebsNecromancy.cfg.txt)

(Lol, just tack a .txt on the end and it uploads just fine. :D )

## To Reproduce (if applicable)

Steps to reproduce the bug. For example:
1. Summon skeleton fire archer.
2. Skeleton blows up (via death from mob or wand smack)
3. Nothing drops.

## Expected behaviour (if applicable)

When skeleton blows up, it should either drop the loot it was crafted with, or it should drop a crate that contains said items, but nothing appears.

**Comment by jpw1991** [comment]
> (Lol, just tack a .txt on the end and it uploads just fine. :D )

100% true, but for people who use Windows and have their file extensions hidden, trying to explain this to them is a true hell lol. So I gave up...

Thanks I'll look at your stuff

**Comment by Zeinath** [comment]
Honestly, I forgot that Windows defaults to hiding extensions... or 'hidden files' for that matter, LOL.

I have a lot of mods going, but no errors currently, which is awesome. The following is the only thing that pops up in the log when I killed the skelly.

'[Info   : Unity Log] 06/26/2023 18:52:17: On death False'

Also, just fyi, there's a mod called 'Pet Protect' that I grabbed to keep my tames from kicking the bucket randomly, which causes the skeletons to just stand there, unable to be interacted with besides hitting with a wand and booming them when they've been killed. I removed that mod, and the issue with the lack of drops is persisting. I tried with both the 'drop to crate' setting on and off to see if it was the crate not spawning that was the issue, but they simply drop nothing when booped.

**Comment by jpw1991** [comment]
Are you sure that the materials dropped aren't being added to an existing crate?

When a creature dies, it checks if there is a crate nearby. If there is, it checks if the crate has room. If it does, it puts the stuff into that existing crate. Otherwise if there is no room, or there is no crate, a new crate is spawned.

**Comment by Zeinath** [comment]
Huh... I was standing next to a tower that likely has a chest in it, will check that really quick. Curious though... with that option disabled, shouldn't the items simply drop on the ground like a normal killed mob?

**Comment by jpw1991** [comment]
> Curious though... with that option disabled, shouldn't the items simply drop on the ground like a normal killed mob?

Yes, correct. If off, it should behave like vanilla. Are all undead affected or just the archers?

**Comment by Zeinath** [comment]
There was a message next to the 'On death false' each time one of them died that was from the Craft From Containers mod. I disabled it, reloaded my client, and the message from that mod is gone, but they still aren't dropping anything when killed, no crate spawning, and nothing going into either my inventory nor the chest just a few steps away.

**Comment by Zeinath** [comment]
Correction, just checked again, Warrior did drop a single bone shard when smacked with wand to kill.

Side question, are they not meant to follow you through portals when following?

**Comment by jpw1991** [comment]
> disabled it, reloaded my client, and the message from that mod is gone, but they still aren't dropping anything when killed, no crate spawning, and nothing going into either my inventory nor the chest just a few steps away.

Plz gimme 15 mins to test this and get back to ya. If it works for me, it's probably interference from another mod. But I don't know which one it could be.

> Side question, are they not meant to follow you through portals when following?

I decided to offload this work to another mod called Teleport Everything, since they already did all the hard work of teleporting creatures and stuff.

**Comment by Zeinath** [comment]
Only mod on my giant list that has anything to do with changes to loot, I believe, is EpicLoot... everything else either messes with the UI, or adds recipes.

**Comment by jpw1991** [comment]
It seems to work just fine for me. Here's a [video](https://youtu.be/FbFrBo9RK7U) I made so you can see the iron bar dropping after the skeleton is destroyed.

On a test server, with a character you don't care about, try turning mods off one by one until it works. Then we'll know which mod is responsible and maybe I can implement some kind of compatibility fix

**Comment by Zeinath** [comment]
Hrm... is it just a chance of some of the materials dropping? Not everything that was used to create them?

**Comment by jpw1991** [comment]
Yes. This is another configurable thing because some people want everything back from a minion, while others want nothing or only the upgrade materials.

So there's the `DropOnDeath` config option that accepts three different values:

- `Nothing` -> minions drop nothing
- `JustResources` -> minions drop just their upgrades. Like in the video - bones weren't dropping, but the iron bar was.
- `Everything` -> minions drop everything that went into creating them

When it comes to `JustResources` and a skeleton archer, the arrows won't be returned. For that, you need to set it to `Everything` and then it should return the bones, the arrows, and the ingot (if it has armor).

```
[UndeadMinion (Server Synced)]
## Whether minions refund resources when they dies.
# Setting type: DropType
# Default value: JustResources
# Acceptable values: Nothing, JustResources, Everything
DropOnDeath = JustResources
```

**Comment by Zeinath** [comment]
Ok, so from your edit, it seems I'm not crazy? I'm getting console unlocked and going to enable debug and all that here in a sec to do more testing that won't be so loss filled. Will let you know if I see anything different with bigger test groups, and maybe putting armor on them (which I couldn't afford to waste resources for, playing solo, friends don't like the game anymore sadly, lol)

**Comment by jpw1991** [comment]
Sure, thanks. If you want to play with people, you have perfect timing because we're just starting up a server. Like, we've set up the mods and the config and it's going live in an hour or two. You're welcome to join us. Please join the discord if you're interested and I'll fill you in

**Comment by Zeinath** [comment]
Just did a bunch of spawning and killing, only with this mod enabled, and yeah, seems when on the 'Everything' setting, it's only dropping one of each for the warriors in armor, and not always a bone fragment for the archers without armor.

**Comment by jpw1991** [comment]
Ok I'll do some debugging and fix it sometime. Thanks for testing & finding the problem

**Comment by jpw1991** [comment]
@Zeinath fixed in 3.6.2
